presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/CommandExecutor.java

```diff
@@ -13,25 +13,79 @@
  */
 package com.teradata.prestomanager.agent;
 
+import com.google.common.io.ByteStreams;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
 import java.util.concurrent.TimeUnit;
 
-import static java.lang.Runtime.getRuntime;
 import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
-// TODO: Add logger
 public final class CommandExecutor
 {
-    private CommandExecutor() {}
+    private static final Logger LOGGER = LogManager.getLogger(CommandExecutor.class);
+    private final String[] commandArray;
+    private int timeout;
+
+    private CommandExecutor(String[] commandArray)
+    {
+        this.commandArray = requireNonNull(commandArray);
+        timeout = 60;
+    }
 
-    public static int executeCommand(String command)
+    public static CommandExecutor commandExecutor(String tokenizeCommand, String... args)
```

Comment:
This isn't a very intuitive breakdown, and I think it isn't needed.   Instead make the other constructor public and pass everything in already tokenized